### PR TITLE
Speculation over the overflow trap on IA32.

### PR DESCRIPTION
**Pull request by asteinha**
Works perfectly on AMD. On Intel it flakes.

### CODE REVIEW

File: demos/instr.h [hunk 1]
```diff
@@ -187,5 +187,45 @@ inline unsigned int ReadUsingES(unsigned int offset) {
...
+
+// Fetches a pointer coming from a non-overflowing operation after an INTO
+// instruction.
+SAFESIDE_ALWAYS_INLINE
+inline void FetchAfterFalseOFCheck(const void *oracle, unsigned int stride,
+                                   const char *data, unsigned int offset) {
+  const char *address = reinterpret_cast<const char *>(oracle);
+  address += stride * data[offset];
+  unsigned int tmp;
+  asm volatile(
```
Comment (mmdriley) [inline demos/instr.h:222]: This assembly sequence also clobbers flags, right?

File: demos/instr.h [hunk 2]
```diff
@@ -187,5 +187,45 @@ inline unsigned int ReadUsingES(unsigned int offset) {
 
   return result;
 }
+
+// Fetches a pointer coming from a non-overflowing operation after an INTO
+// instruction.
+SAFESIDE_ALWAYS_INLINE
+inline void FetchAfterFalseOFCheck(const void *oracle, unsigned int stride,
```
Comment (mmdriley) [inline demos/instr.h]: Is there a way we could fold these (this function and the next) together into one that takes arguments from the caller that might or might not cause overflow? Something like `SupposedlySafeOffsetAndDereference()` that takes an address and an offset, adds them, checks overflow, then deferences.

First call: `SSOAD(address, 5)`
Second, exploit call: `SSOAD(address + 1000, INT_MAX - 1000 + 5)`

**Review by mmdriley — APPROVED** [review]

File: demos/instr.h (hunk 1 above)
Comment (asteinha) [reply demos/instr.h:222]: Yes. Obsolete.

File: demos/instr.h (hunk 2 above)
Comment (asteinha) [reply demos/instr.h]: Done.

**Pull request merged**